dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -180,18 +180,16 @@ bool hasRegionToRead(const DAGContext & dag_context, const TiDBTableScan & table
 // <has_cast, extra_cast, project_for_remote_read>
 std::pair<bool, ExpressionActionsPtr> addExtraCastsAfterTs(
     DAGExpressionAnalyzer & analyzer,
-    const std::vector<ExtraCastAfterTSMode> & need_cast_column,
+    const std::vector<bool> & may_need_add_cast_column,
```

Comment:
`vector<bool>` is a special specialization for `vector`, it could brings unexpected behavior sometimes. Maybe use `vector<UInt8>` instead?

Reply:
The special case is
```
# vec[i] = true;
auto b = vec[i];
b = false.
# vec[i] become false now.
```

We will not do this way, so it is safe.

Reply:
We won't do it now and we know this problem. But it could bring trouble in further code changes who don't familiar with `vector<bool>`.
Compare to `vector<UInt8>`, I think `vector<bool>` here does not bring any special benefit.